SonarQube cannot start due to JVM creation error

**`brew config` output**

HOMEBREW_VERSION: 2.7.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: cd191a7f6c4c8e053336d0ea34a9886b12d06470
Last commit: 5 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9d698bde1b0369557cfa462dcc07d6168088c67a
Core tap last commit: 88 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 11.0 build 1103
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 11.0.10, 1.8.0_282
macOS: 10.15.7-x86_64
CLT: 1103.0.32.62
Xcode: N/A

**`brew doctor` output**

Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/more/.

****

- [X] The `brew doctor` above contains no "Warning" lines.

**What were you trying to do (and why)?**

I'm trying to run SonarQube using Homebrew

NOTE: My organization currently prevents me from updating from updating to the just-released CLT update. I'm running clang version 1103.0.32.62 instead of the latest 1200.0.32.27.

**What happened (include all command output)?**

When I start up SonarQube (either background or foreground), it fails due to a JVM creation error (below). It appears that the SonarQube formal depends on OpenJDK 15.0.1 (https://formulae.brew.sh/formula/sonarqube#default), but that JDK version does not support a memory management argument SonarQube uses (UseConcMarkSweepGC). The officially supported OpenJDK version for SonarQube is 11: https://docs.sonarqube.org/latest/requirements/requirements/#header-3. I suspect this issue would also affect the sonarqube-lts formula since it uses the same OpenJDK dependency.

> sonar console
Running SonarQube...
wrapper  | --> Wrapper Started as Console
wrapper  | Launching a JVM...
jvm 1    | Wrapper (Version 3.2.3) http://wrapper.tanukisoftware.org
jvm 1    |   Copyright 1999-2006 Tanuki Software, Inc.  All Rights Reserved.
jvm 1    | 
jvm 1    | 2021.02.02 09:21:12 INFO  app[][o.s.a.AppFileSystem] Cleaning or creating temp directory /usr/local/Cellar/sonarqube/8.6.1.40680/libexec/temp
jvm 1    | 2021.02.02 09:21:12 INFO  app[][o.s.a.es.EsSettings] Elasticsearch listening on [HTTP: 127.0.0.1:9001, TCP: 127.0.0.1:52876]
jvm 1    | 2021.02.02 09:21:12 INFO  app[][o.s.a.ProcessLauncherImpl] Launch process[[key='es', ipcIndex=1, logFilenamePrefix=es]] from [/usr/local/Cellar/sonarqube/8.6.1.40680/libexec/elasticsearch]: /usr/local/Cellar/sonarqube/8.6.1.40680/libexec/elasticsearch/bin/elasticsearch
jvm 1    | 2021.02.02 09:21:12 INFO  app[][o.s.a.SchedulerImpl] Waiting for Elasticsearch to be up and running
jvm 1    | Unrecognized VM option 'UseConcMarkSweepGC'
jvm 1    | Error: Could not create the Java Virtual Machine.
jvm 1    | Error: A fatal exception has occurred. Program will exit.
jvm 1    | 2021.02.02 09:21:12 WARN  app[][o.s.a.p.AbstractManagedProcess] Process exited with exit value [es]: 1
jvm 1    | 2021.02.02 09:21:12 INFO  app[][o.s.a.SchedulerImpl] Process[es] is stopped
jvm 1    | 2021.02.02 09:21:12 INFO  app[][o.s.a.SchedulerImpl] SonarQube is stopped
jvm 1    | 2021.02.02 09:21:12 ERROR app[][o.s.a.p.EsManagedProcess] Failed to check status
jvm 1    | java.lang.RuntimeException: thread waiting for the response was interrupted
jvm 1    | 	at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:818)
jvm 1    | 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:259)
jvm 1    | 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:246)
jvm 1    | 	at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1613)
jvm 1    | 	at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1583)
jvm 1    | 	at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1553)
jvm 1    | 	at org.elasticsearch.client.ClusterClient.health(ClusterClient.java:130)
jvm 1    | 	at org.sonar.application.es.EsConnectorImpl.getClusterHealthStatus(EsConnectorImpl.java:56)
jvm 1    | 	at org.sonar.application.process.EsManagedProcess.checkStatus(EsManagedProcess.java:89)
jvm 1    | 	at org.sonar.application.process.EsManagedProcess.checkOperational(EsManagedProcess.java:74)
jvm 1    | 	at org.sonar.application.process.EsManagedProcess.isOperational(EsManagedProcess.java:59)
jvm 1    | 	at org.sonar.application.process.ManagedProcessHandler.refreshState(ManagedProcessHandler.java:220)
jvm 1    | 	at org.sonar.application.process.ManagedProcessHandler$EventWatcher.run(ManagedProcessHandler.java:285)
jvm 1    | Caused by: java.lang.InterruptedException: null
jvm 1    | 	at java.base/java.lang.Object.wait(Native Method)
jvm 1    | 	at java.base/java.lang.Object.wait(Object.java:321)
jvm 1    | 	at org.apache.http.concurrent.BasicFuture.get(BasicFuture.java:82)
jvm 1    | 	at org.apache.http.impl.nio.client.FutureWrapper.get(FutureWrapper.java:70)
jvm 1    | 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:255)
jvm 1    | 	... 11 common frames omitted
wrapper  | <-- Wrapper Stopped

**What did you expect to happen?**

SonarQube should've finished starting up

**Step-by-step reproduction instructions (by running `brew` commands)**

brew update
brew install sonarqube
sonar console

--- a/Formula/sonarqube.rb
+++ b/Formula/sonarqube.rb
@@ -4,6 +4,7 @@ class Sonarqube < Formula
   url "https://binaries.sonarsource.com/Distribution/sonarqube/sonarqube-8.6.1.40680.zip"
   sha256 "e16a4208658c485247f06f9b79c2d3b060e86751b35e95adef9f881754ce3e1d"
   license "LGPL-3.0-or-later"
+  revision 1
 
   livecheck do
     url "https://binaries.sonarsource.com/Distribution/sonarqube/"
@@ -12,7 +13,7 @@ class Sonarqube < Formula
 
   bottle :unneeded
 
-  depends_on "openjdk"
+  depends_on "openjdk@11"
 
   conflicts_with "sonarqube-lts", because: "both install the same binaries"
 
@@ -23,7 +24,7 @@ class Sonarqube < Formula
     libexec.install Dir["*"]
 
     (bin/"sonar").write_env_script libexec/"bin/macosx-universal-64/sonar.sh",
-      JAVA_HOME: Formula["openjdk"].opt_prefix
+      Language::Java.overridable_java_home_env("11")
   end
 
   plist_options manual: "sonar console"

